io.openems.edge.goodwe.et/src/io/openems/edge/goodwe/et/FeneconGoodweET.java

```diff
@@ -0,0 +1,1104 @@
+	private boolean readonly = false;
+
+	public static final int DEFAULT_UNIT_ID = 5;
+	protected final static int MAX_APPARENT_POWER = 100_00;
+
+	// private Battery battery;
+
+	@Reference
+	protected ConfigurationAdmin cm;
+
+	@Reference
+	private Power power;
+
+	@Reference(policy = ReferencePolicy.STATIC, policyOption = ReferencePolicyOption.GREEDY, cardinality = ReferenceCardinality.MANDATORY)
+	protected void setModbus(BridgeModbus modbus) {
+		super.setModbus(modbus); // Bridge Modbus
+	}
+
+	@Activate
+	void activate(ComponentContext context, Config config) {
+		super.activate(context, config.id(), config.alias(), config.enabled(), config.unit_id(), this.cm, "Modbus",
+				config.modbus_id());
+
+		this.readonly = config.readonly();
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		super.deactivate();
+	}
+
+	public FeneconGoodweET() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				SymmetricEss.ChannelId.values(), //
+				AsymmetricEss.ChannelId.values(), //
+				GoodweChannelIdET.values() // //
+		);
+		this.channel(SymmetricEss.ChannelId.MAX_APPARENT_POWER).setNextValue(FeneconGoodweET.MAX_APPARENT_POWER);
+	}
+
+	@Override
+	protected ModbusProtocol defineModbusProtocol() {
+		return new ModbusProtocol(this,
+				new FC3ReadRegistersTask(35000, Priority.LOW,
+						m(GoodweChannelIdET.MODBUS_PROTOCOL_VERSION, new UnsignedWordElement(35000)), //
+						m(GoodweChannelIdET.RATED_POWER, new UnsignedWordElement(35001)), //
+						m(GoodweChannelIdET.AC_OUTPUT_TYPE, new UnsignedWordElement(35002))), //
+				new FC3ReadRegistersTask(35003, Priority.LOW,
+						m(GoodweChannelIdET.SERIAL_NUMBER, new StringWordElement(35003, 8)), //
+						m(GoodweChannelIdET.DEVICE_TYPE, new StringWordElement(35011, 5))), //
+				new FC3ReadRegistersTask(35016, Priority.LOW,
+						m(GoodweChannelIdET.DSP1_SOFTWARE_VERSION, new UnsignedWordElement(35016)), //
+						m(GoodweChannelIdET.DSP2_SOFTWARE_VERSION, new UnsignedWordElement(35017)), //
+						m(GoodweChannelIdET.DSP_SPN_VERSION, new UnsignedWordElement(35018)), //
+						m(GoodweChannelIdET.ARM_SOFTWARE_VERSION, new UnsignedWordElement(35019)), //
+						m(GoodweChannelIdET.ARM_SVN_VERSION, new UnsignedWordElement(35020))), //
+				new FC3ReadRegistersTask(35021, Priority.LOW,
+						m(GoodweChannelIdET.DSP_INTERNAL_FIRMWARE_VERSION, new StringWordElement(35021, 6))), //
+				new FC3ReadRegistersTask(35027, Priority.LOW,
+						m(GoodweChannelIdET.ARM_INTERNAL_FIRMWARE_VERSION, new StringWordElement(35027, 6))), //
+				new FC3ReadRegistersTask(35050, Priority.LOW,
+						m(GoodweChannelIdET.SIMCCID, new StringWordElement(35050, 10))), //
+				new FC3ReadRegistersTask(35100, Priority.LOW,
+						m(GoodweChannelIdET.RTC_YEAR_MONTH, new UnsignedWordElement(35100)), //
+						m(GoodweChannelIdET.RTC_DATE_HOUR, new UnsignedWordElement(35101)), //
+						m(GoodweChannelIdET.RTC_MINUTE_SECOND, new UnsignedWordElement(35102))), //
+				new FC3ReadRegistersTask(35103, Priority.HIGH,
```

Comment:
Why is this HIGH priority? Try to have as few HIGH priority tasks as possible, as they are forced to be executed on every cycle.